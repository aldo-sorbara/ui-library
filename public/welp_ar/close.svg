<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
