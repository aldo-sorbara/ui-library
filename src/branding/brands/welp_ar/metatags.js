export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
