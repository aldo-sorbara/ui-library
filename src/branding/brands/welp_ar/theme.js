import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
