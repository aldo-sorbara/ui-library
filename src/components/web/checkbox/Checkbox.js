import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/**
 * @render react
 * @name Checkbox
 * @description Checkbox Component for Wenance
 * @example
 * <Checkbox data="checkbox"> This is a checkbox</Checkbox>
 */

const IconCheck = props => (
  <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 512 512'>
    <path
      fill='currentColor'
      d='M435.848 83.466L172.804 346.51l-96.652-96.652c-4.686-4.686-12.284-4.686-16.971 0l-28.284 28.284c-4.686
      4.686-4.686 12.284 0 16.971l133.421 133.421c4.686 4.686 12.284 4.686 16.971 0l299.813-299.813c4.686-4.686
      4.686-12.284 0-16.971l-28.284-28.284c-4.686-4.686-12.284-4.686-16.97 0z'
    />
  </svg>
);

const Checkbox = props => {
  const { children, data, ...rest } = props;
  return (
    <CheckboxContainer>
      <CheckboxLabel>
        <input data-w={data} {...rest} {...rest.input} type='checkbox' />
        <span>
          <IconCheck />
        </span>

        {children}
      </CheckboxLabel>
    </CheckboxContainer>
  );
};

Checkbox.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  data: PropTypes.string.isRequired
};

const CheckboxContainer = styled.div`
  font-size: 16px;
  position: relative;

  input[type='checkbox'] {
    height: 1.25rem;
    left: 0;
    margin: 0;
    opacity: 0;
    position: absolute;
    top: 0;
    width: 1.25rem;
    z-index: ${props => props.theme.zIndex.checkbox};
  }

  span:nth-of-type(1) {
    background-color: transparent;
    border: 1px solid ${props => props.theme.palette.primary.default};
    border-radius: 4px;
    color: ${props => props.theme.palette.common.bodyBg};
    height: 1.25rem;
    left: 0;
    position: absolute;
    top: ${props => props.theme.spacer.small};
    transition: border-color, background-color 0.2s ease-in-out;
    width: 1.25rem;
  }

  input:checked {
    & + span {
      background-color: ${props => props.theme.palette.primary.default};
      padding: 1px;

      &::after {
        content: '';
        display: block;
      }
    }
  }
`;

const CheckboxLabel = styled.label`
  display: block;
  ${props => props.theme.typography.checkbox};
  padding-bottom: ${props => props.theme.spacer.small};
  padding-left: 2rem;
  padding-top: ${props => props.theme.spacer.small};
`;

CheckboxContainer.defaultProps = {
  theme: {
    zIndex: {
      checkbox: 1
    },
    palette: {
      common: {
        bodyBg: '#ffffff'
      },
      primary: {
        default: '#0082ff'
      }
    },
    typography: {
      checkbox: {
        'font-size': '1rem',
        'font-weight': '400',
        'line-height': '1.25'
      }
    },
    spacer: {
      small: '.5rem'
    }
  }
};

export default Checkbox;
