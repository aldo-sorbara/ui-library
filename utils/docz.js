export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
