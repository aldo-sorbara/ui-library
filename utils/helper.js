import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
