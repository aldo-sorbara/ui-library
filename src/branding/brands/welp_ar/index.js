import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
