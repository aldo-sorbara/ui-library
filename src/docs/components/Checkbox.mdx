---
name: Checkbox
menu: Components
route: /checkbox
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Checkbox from '../../components/web/checkbox';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# Checkbox

<PropsTable of={Checkbox} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Checkbox
        data='checkbox'
        onChange={e => {
          console.log(e.target.checked);
        }}
      >
        This is a checkbox.
      </Checkbox>
      <Checkbox
        data='checkbox2'
        onChange={e => {
          console.log(e.target.checked);
        }}
      >
        This is another <a href=''>checkbox with link </a>
      </Checkbox>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/checkbox--with-default-props)
