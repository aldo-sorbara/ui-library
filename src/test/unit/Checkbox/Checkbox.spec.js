import React from 'react';
import Checkbox from '../../../components/web/checkbox';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../../branding/brands/welp_ar';
import 'jest-styled-components';

describe('Checkbox component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Checkbox className='test-checkbox' data='checkbox'>
          Checkbox
        </Checkbox>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render checked checkbox', () => {
    const wrapper = mountWithTheme(
      <Checkbox
        className='test-checkbox'
        checked
        onChange={() => {
          console.log('test');
        }}
        data='checkbox'
      >
        Checked
      </Checkbox>,
      welp_ar.theme
    );
    expect(wrapper.props().checked).toBe(true);
  });
  it('should render checkbox with default props', () => {
    const wrapper = mountWithTheme(
      <Checkbox className='test-checkbox' data='checkbox'>
        checkbox
      </Checkbox>,
      welp_ar.theme
    );
    expect(wrapper.props().data).toBe('checkbox');
    expect(wrapper.props().children).toBe('checkbox');
  });
});
