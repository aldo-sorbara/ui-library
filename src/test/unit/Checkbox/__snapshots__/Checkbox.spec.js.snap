// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render snapshot 1`] = `
.c0 {
  font-size: 16px;
  position: relative;
}

.c0 input[type='checkbox'] {
  height: 1.25rem;
  left: 0;
  margin: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  width: 1.25rem;
  z-index: 1;
}

.c0 span:nth-of-type(1) {
  background-color: transparent;
  border: 1px solid #0082ff;
  border-radius: 4px;
  color: #ffffff;
  height: 1.25rem;
  left: 0;
  position: absolute;
  top: .5rem;
  -webkit-transition: border-color,background-color 0.2s ease-in-out;
  transition: border-color,background-color 0.2s ease-in-out;
  width: 1.25rem;
}

.c0 input:checked + span {
  background-color: #0082ff;
  padding: 1px;
}

.c0 input:checked + span::after {
  content: '';
  display: block;
}

.c1 {
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.25;
  padding-bottom: .5rem;
  padding-left: 2rem;
  padding-top: .5rem;
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <input
      className="test-checkbox"
      data-w="checkbox"
      type="checkbox"
    />
    <span>
      <svg
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M435.848 83.466L172.804 346.51l-96.652-96.652c-4.686-4.686-12.284-4.686-16.971 0l-28.284 28.284c-4.686 4.686-4.686 12.284 0 16.971l133.421 133.421c4.686 4.686 12.284 4.686 16.971 0l299.813-299.813c4.686-4.686 4.686-12.284 0-16.971l-28.284-28.284c-4.686-4.686-12.284-4.686-16.97 0z"
          fill="currentColor"
        />
      </svg>
    </span>
    Checkbox
  </label>
</div>
`;
